src/new/wallet.js

```diff
@@ -505,6 +500,9 @@ class HathorWallet extends EventEmitter {
    * @inner
    */
   getAddressAtIndex(index) {
+    if (this.preCalculatedAddresses && this.preCalculatedAddresses[index]) {
```

Comment:
I think we should think this more thoroughly, there are possible issues, e.g. we have an array with 20 addresses and decide to load the index 100k, the "cache" array will need to be recreated to hold the first 20 addresses and a lot of `undefined`s up to the 100k index.

Although I think we should minimize the derivation operations since they are cpu intensive there is also a tradeoff of increasing the wallet's memory requirements.